### ISSUE TITLE
BUG: scipy.sparse.linalg.expm_multiply unstable for start != 0

### ISSUE THREAD

**Issue by pdcook** [scipy/scipy#16363]
### Describe your issue.

This is a re-hash of #16311, with hopefully a more correct understanding of what the issue actually is and a significantly better example.

I am able to replicate this issue with both v1.8.0 and v1.9.0.dev0.

### Goal
Compute the matrix exponential of `u * A` where `u` is some scalar and `A` is a square matrix (dense, sparse, or `LinearOperator`). For a range of values for `u`, not necessarily starting at or near `0`.

### Issue
`scipy.linalg.expm_multiply` is the exact tool for this, and offers `start`, `stop`, `num`, and `endpoint` options which it will internally create a `linspace` from for `u` and then compute `expm(u*A).dot(<some other matrix>)` for all of the values of `u` in that `linspace`. However, for `start != 0` the error in the result is significantly larger than it otherwise would be for the same inputs but with `start = 0`.

This issue is entirely limited to `scipy.linalg.expm_multiply` with `start != 0`. The following functions produce results with errors consistent with the Pade approximation and machine precision:
- Diagonalizing `A = V E V^(-1)` and computing `V @ np.diag(np.exp(E)) @ np.inv(V)`
- `scipy.sparse.linalg.expm(u * A)`
- `np.matrix_power(scipy.sparse.linalg.expm(du * A), num)` where `du = u / num`
- `scipy.sparse.linalg.expm_multiply(u*A, np.identity(A.shape[0]))`
- `scipy.sparse.linalg.expm_multiply(A, np.identity(A.shape[0]), start = 0, stop = u, num = num, endpoint=True)[~0]` (taking the last result in the sequence of `u`s
- The above with values of `start << u` (`start` much smaller than `u`)

### Testing

Using the 3x3 matrix
```
A = [[ 0, 1, 0],
        [ 1, 0, 1],
        [ 0, 1, 0]]
```
which one can "easily" compute `expm(u * A)` analytically by hand in terms of `sinh` and `cosh`, I compared the above methods and `scipy.sparse.linalg.expm_multiply` with `start != 0` for several values of `u`. As mentioned above, all methods except `scipy.sparse.linalg.expm_multiply` with `start` sufficiently far from `0` were consistent. See figure:

![expmerror](https://user-images.githubusercontent.com/16090923/172248971-f4b22d36-85ab-49b1-b76e-e1f1dc2ffd02.png)

The code to reproduce this figure is below.

### Reproducing Code Example

```python
import numpy as np
import scipy.sparse.linalg as spla
import itertools
import matplotlib.pyplot as plt


def Test(u, num):

    """
        Test matrix
    """
    A = np.array([[0, 1, 0], [1, 0, 1], [0, 1, 0]], dtype=np.double)

    """
        Identity matrix, needed for expm_multiply
    """
    B = np.identity(A.shape[0])

    """
        Exact analytic solution
    """
    sqrt2 = np.sqrt(2)
    eA_exact = np.array(
        [
            [
                np.cosh(u / sqrt2) ** 2,
                np.sinh(sqrt2 * u) / sqrt2,
                np.sinh(u / sqrt2) ** 2,
            ],
            [
                np.sinh(sqrt2 * u) / sqrt2,
                np.cosh(sqrt2 * u),
                np.sinh(sqrt2 * u) / sqrt2,
            ],
            [
                np.sinh(u / sqrt2) ** 2,
                np.sinh(sqrt2 * u) / sqrt2,
                np.cosh(u / sqrt2) ** 2,
            ],
        ]
    )

    """
        Solution by diagonalization
    """
    E, V = np.linalg.eig(A)
    i = np.argsort(E)
    E = E[i]
    V = V[:, i]
    eA_exact_diagonalization = V @ np.diag(np.exp(E * u)) @ np.linalg.inv(V)

    """
        Solution by expm
    """
    eA_direct = spla.expm(u * A)

    """
        Solution by computing expm(du*A) and then exponentiating the result
    """
    du = u / num
    eA_steps_manual = np.linalg.matrix_power(spla.expm(du * A), num)

    """
        Solution by expm_multiply without start or stop
    """
    eA_multiply_direct = spla.expm_multiply(u * A, B)

    """
        Solutions by expm_multiply WITH start and stop for various values of
        start that are successively closer to stop
    """
    eA_multiply_steps = spla.expm_multiply(
        A, B, start=0, stop=u, num=num, endpoint=True
    )[~0]
    eA_multiply_steps_quarter = spla.expm_multiply(
        A, B, start=0.25 * u, stop=u, num=num, endpoint=True
    )[~0]
    eA_multiply_steps_half = spla.expm_multiply(
        A, B, start=0.5 * u, stop=u, num=num, endpoint=True
    )[~0]
    eA_multiply_steps_three_quarter = spla.expm_multiply(
        A, B, start=0.75 * u, stop=u, num=num, endpoint=True
    )[~0]
    eA_multiply_steps_late = spla.expm_multiply(
        A, B, start=0.9 * u, stop=u, num=num, endpoint=True
    )[~0]
    eA_multiply_steps_verylate = spla.expm_multiply(
        A, B, start=0.99 * u, stop=u, num=num, endpoint=True
    )[~0]

    # compute frobenius norm between every solution and the exact solution
    solutions = {
        "exact": eA_exact,
        "V e^{u\Lambda} V^{-1}": eA_exact_diagonalization,
        "expm(uA)": eA_direct,
        "matrix_power(expm(duA), %d)" % num: eA_steps_manual,
        "expm_multiply(uA, I)": eA_multiply_direct,
        "expm_multiply(A, I, 0, u, %d)" % num: eA_multiply_steps,
        "expm_multiply(A, I, 0.25u, u, %d)" % num: eA_multiply_steps_quarter,
        "expm_multiply(A, I, 0.5u, u, %d)" % num: eA_multiply_steps_half,
        "expm_multiply(A, I, 0.75u, u, %d)" % num: eA_multiply_steps_three_quarter,
        "expm_multiply(A, I, 0.9u, u, %d)" % num: eA_multiply_steps_late,
        "expm_multiply(A, I, 0.99u, u, %d)" % num: eA_multiply_steps_verylate,
    }

    exact_vs = {}
    for name, sol in solutions.items():
        if name == "exact":
            continue
        exact_vs[name] = np.linalg.norm(sol - eA_exact, ord="fro")

    return exact_vs


def PlotTest(umax, num_us, num_steps):
    fig, ax = plt.subplots(1, 1)

    u = np.linspace(0, umax, num_us)

    exact_vs = {}

    for i, ui in enumerate(u):
        vs = Test(ui, num_steps)
        for k, v in vs.items():
            exact_vs[k] = exact_vs.get(k, []) + [v]

    for k, v in exact_vs.items():
        ax.semilogy(
            u,
            v,
            label=(r"$\mathtt{%s}$" % k.replace("_", "\_")),
            lw=3 if "multiply" in k else 2,
            ls="-" if "multiply" in k else "--",
        )
    ax.legend(fontsize=15, loc="upper left", ncol=2)
    ax.set_title(
        r"$\left|\left|e^{uA}_{\mathrm{numerical}}-e^{uA}_{\mathrm{exact}}\right|\right|_F$"
    )

    plt.show()

if __name__ == "__main__":
    PlotTest(20, 100, 100)
```

### Error message

```shell
N/A
```

### SciPy/NumPy/Python version information

1.8.0 1.21.0 sys.version_info(major=3, minor=8, micro=10, releaselevel='final', serial=0) | 1.9.0.dev0 1.22.4 sys.version_info(major=3, minor=8, micro=10, releaselevel='final', serial=0)

**Comment by pdcook** [comment]
I believe I have tracked down this error in it's entirety.

#### TLDR: 

[Line 699 in `_expm_multiply.py`](https://github.com/scipy/scipy/blob/7f941dbc76ad90aa4dc23b613f5c2da454115075/scipy/sparse/linalg/_expm_multiply.py#L699) is 
```
    X[0] = _expm_multiply_simple_core(A, B, t_0, mu, m_star, s)
```
but it should be
```
    if t_0*A_1_norm == 0:
        m_star_0, s_0 = 0, 1
    else:
        norm_info_0 = LazyOperatorNormInfo(t_0*A, A_1_norm=t_0*A_1_norm, ell=ell)
        m_star_0, s_0 = _fragment_3_1(norm_info_0, n0, tol, ell=ell)
    X[0] = _expm_multiply_simple_core(A, B, t_0, mu, m_star_0, s_0)
```

#### Explanation

[The article used as the source for the algorithm](http://eprints.maths.manchester.ac.uk/1591/1/alhi11.pdf) requires the computation of two constants, `mu_star` and `s` which depend on (among other things) the `LazyOperatorNormInfo` of `t * A`.

For `_expm_multiply_interval` this `t` is defined as `samples[~0] - samples[0]` (the "width" of the interval). However, for `_expm_multiply_simple` this `t` is simply the `t` at which `expm(t * A)` is being calculated. When `start != 0`, `expm_multiply` will use `_expm_multiply_simple` with `t = t_0 = samples[0]`, _but will still use the same `mu_star` and `s` calculated for the whole interval from `start` to `stop`.

This explains perfectly why in my example, `expm_multiply` produced accurate results for `start <= 0.5 * stop`. Since in this case `t_0 <= samples[~0] - samples[0]`. So in reality this bug isn't for `start != 0` but instead for `start > stop / 2`.

#### Results With Fix

![expm_fixed](https://user-images.githubusercontent.com/16090923/172475727-2f93cc97-7b08-4eca-876a-17c15f04ce55.png)
